Refactor Into python module and CLI

jkoelker opened the pull request:
- Switches to `argparse` for command line processing.
- Move cli script to `entry_points['console_scripts']`
- Separate display and module logic
- Add token caching for faster use in cron scripts

Closes #3

jkoelker committed:
General Cleanup and Argparse switch

Use argparse for command line
Convert command line to console_scripts entry_point
PEP8 everything

jkoelker committed:
Major refactor

Is now a proper python library for long running use
Command line is converted to use this

jkoelker committed:
Remove the unused Nest._set function

All writing is happening in either Device or Structure classes

jkoelker committed:
Use common NestBase object

DRY up Device/Structure a bit

jkoelker committed:
Remove use of old self.login()

jkoelker committed:
Functions were named wrong ;)

jkoelker committed:
dont' set the name to the serial on instatiation ;)

jkoelker committed:
Look at device mode instead of args

jkoelker committed:
Add config file support

Prevent username/password from being on the ps line

jkoelker committed:
Add token caching

Allow the access_token (and other auth data like URLS) to be cached
inbetween calls. Makes calling from one shot scripts faster and more
better.

jkoelker committed:
Run the auth_callback for cached data

When using the cached data, make sure to call the auth_callback

jkoelker committed:
Remove the stray debugging print ;)

jkoelker committed:
Fix cli help

jkoelker committed:
Update README.md

jkoelker committed:
Add Weather support

jkoelker committed:
Use `decimal.Decimal` for conversions

Helps to ensure precision

jkoelker committed:
Update README.md

jkoelker committed:
Update README.md

jkoelker committed:
`gmt_offset` in NestTZ should be a float

jkoelker committed:
should be hours

jkoelker committed:
unindent ;)

jkoelker committed:
Provide a __repr__ for NestTZ

kjmancuso commented:
So.....

FiloSottile commented:
@jkoelker Whoops, sorry, I first encouraged a PR and then left for a vacation :(

I'll review the changes and get back to you ASAP, thanks for your work!

fxstein commented:
Any updates?

bsmithyman committed:
Improve handling of 'celsius' option via config

Celsius units can't be set via config file because of hard-coded default after the config is read in. Move hard-coded default to the initialization of the dictionary, and update the dictionary based on the config file.

jkoelker committed:
Merge pull request #1 from bsmithyman/master

Improve handling of 'celsius' option via config

fxstein committed:
Add temperature conversion to show option

Convert all temperatures as displayed by show to the selected scale
(C/F)

fxstein committed:
Update README install

Update install section of README to perform local fork based install as
pip will always install an older version

fxstein committed:
Additional whitespaces

jkoelker committed:
Merge pull request #2 from fxstein/master

Apply temperature scales to show option

logjames committed:
Target Humidity Functions

Added functionality to set the target humidity with either a specified
value, or by auto-selecting an appropriate humidity value based on the
Weather API current temperature. The following auto-selection rules are
used:
                     ￼Outdoor Temperature (Degrees Fahrenheit)     Set
Target Humidity (%)
                         +40°
45%
                         +30°
40%
                         +20°                                        35%
                         +10°
30%
                           0°
25%
                         -10°
20%
                         -20°                                        15%

jkoelker committed:
Merge pull request #3 from logjames/master

Target Humidity Functions

balloob committed:
Make nest.py Python3 compatible

jkoelker committed:
Merge pull request #4 from balloob/master

Make nest.py Python3 compatible

jkoelker committed:
Fix up code style to PEP8

jkoelker committed:
Add travis-ci

jkoelker committed:
Add Build Status

jkoelker committed:
Change the flask8 module ;)

jkoelker committed:
Fix comments

jkoelker committed:
Fix python 3 (print)

jkoelker committed:
Rename for pypi

The python module is now named `nest` which pyth pypi name is
`python-nest`.

jkoelker committed:
Update nest.py preamble

jkoelker committed:
Fix Readme for PyPi display

jkoelker committed:
Fix .travis.yaml

balloob committed:
Make setup.py work on Python 3

jkoelker committed:
Update README.rst to use new build svg

logjames committed:
Update README.rst

Fixed formatting of the command line

jkoelker committed:
Fix console scripts

Missed the rename in console scripts. Also fix the last reminent of the
old `nest_thermostat` in README.rst

jkoelker committed:
Fix temperature setting when mode != range

Since args.temperature is always a list, when mode is not range we still
need to use args.temperature[0]

Fixes #6

jkoelker committed:
Disable token caching from the cmd line

Token caching is currently not working. Disable for now. Filed #7 issue
to fix.

jkoelker committed:
Bump release to v2.3

jkoelker committed:
Allow specifying serial or structure for away

Adds the global -S/--structure argument for specifying the structure
name. Away status will be scoped to the given structure name or to the
structure that a device appears in, falling back to just the first
structure found.

Also scopes device discovery for all other functions if specifed to only
the devices in that structure.

Fixes #8

jkoelker committed:
Bump to v2.3.1
